### PR TITLE
replaces dependabot with renovate

### PR DESCRIPTION
**Pull request by esacteksab**
As a part of this [effort](https://trello.com/c/RjhTGJ0Y/155-migrate-dependabot-to-renovatebot).

This also removes Kodiak, updates pre-commit and formats as a result. Also updates Go to 1.19, and subsequent dependencies. Also `io/ioutil` was deprecated in 1.16, so fixed that as well.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
 
+  - repo: https://github.com/executablebooks/mdformat
+    rev: 0.7.16
+    hooks:
+      - id: mdformat
+        additional_dependencies:
+          - mdformat-gfm
+          - mdformat-toc
+        # mdformat fights with terraform_docs
+        exclude: README.m(ark)?d(own)?
```
Comment (rpdelaney) [inline .pre-commit-config.yaml]:
```suggestion
        # mdformat fights with terraform_docs
        exclude: README.m(ark)?d(own)?
```
No terraform here so I think we can let mdformat do all the things

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      "managers": [
+        "terraform",
+        "pre-commit",
+        "gomod",
+        "dockerfile"
+      ],
+      "matchUpdateTypes": [
+        "minor",
+        "patch"
+      ]
```
Comment (rpdelaney) [inline renovate.json]: Nit: What about pin and digest?

**Review by rpdelaney — APPROVED** [review]
Conditional approval: Please consider my comments and apply if you agree, or merge this PR as-is if you don't :)

File: renovate.json (hunk 2 above)
Comment (rpdelaney) [reply renovate.json]: I'm actually not sure what this is doing that isn't already accomplished from lines 7-13

File: .pre-commit-config.yaml (hunk 1 above)
Comment (esacteksab) [reply .pre-commit-config.yaml]: You're correct. Pushed a fix.

File: renovate.json (hunk 2 above)
Comment (esacteksab) [reply renovate.json]: Bad copy/paste. Pushed a fix. Let me know.

File: renovate.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        "terraform",
+        "pre-commit",
+        "gomod",
+        "dockerfile"
+      ],
+      "matchUpdateTypes": [
+        "minor",
+        "patch",
+        "pin",
+        "patch"
```
Comment (rpdelaney) [inline renovate.json]:
```suggestion
        "digest"
```

**Review by rpdelaney — APPROVED** [review]
That looks great! One small nit left :) Thank you!

**Comment by esacteksab** [comment]
> That looks great! One small nit left :) Thank you!

No, no! Thank you!!

**Pull request merged**